### PR TITLE
Add support for KtLint 0.48.0 (fixes #1430)

### LINKED ISSUE #1430: Spotless doesn't work with ktlint 0.48.0
If you are submitting a **bug**, please include the following:

- [ 6.12.0 ] spotless version

This is stack trace:
```
java.lang.NoSuchMethodError: 'com.pinterest.ktlint.core.api.UsesEditorConfigProperties$EditorConfigProperty com.pinterest.ktlint.core.api.DefaultEditorConfigProperties.getKtlintDisabledRulesProperty()'
        at com.diffplug.spotless.glue.ktlint.compat.KtLintCompat0Dot47Dot0Adapter.createEditorConfigOverride(KtLintCompat0Dot47Dot0Adapter.java:106)
        at com.diffplug.spotless.glue.ktlint.compat.KtLintCompat0Dot47Dot0Adapter.format(KtLintCompat0Dot47Dot0Adapter.java:73)
        at com.diffplug.spotless.glue.ktlint.KtlintFormatterFunc.applyWithFile(KtlintFormatterFunc.java:71)
        at com.diffplug.spotless.FormatterFunc$NeedsFile.apply(FormatterFunc.java:154)
        at com.diffplug.spotless.FormatterStepImpl$Standard.format(FormatterStepImpl.java:82)
        at com.diffplug.spotless.FormatterStep$Strict.format(FormatterStep.java:88)
        at com.diffplug.spotless.Formatter.compute(Formatter.java:230)
        at com.diffplug.spotless.PaddedCell.calculateDirtyState(PaddedCell.java:203)
        at com.diffplug.spotless.PaddedCell.calculateDirtyState(PaddedCell.java:190)
        at com.diffplug.gradle.spotless.SpotlessTaskImpl.processInputFile(SpotlessTaskImpl.java:102)
        at com.diffplug.gradle.spotless.SpotlessTaskImpl.performAction(SpotlessTaskImpl.java:88)
```

### PR DESCRIPTION
**Pull request by davidburstrom**

### CODE REVIEW

File: lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	 */
+	private static EditorConfigOverride createEditorConfigOverride(final List<Rule> rules, Map<String, Object> editorConfigOverrideMap) {
+		// Get properties from rules in the rule sets
+		Stream<EditorConfigProperty<?>> ruleProperties = rules.stream()
+				.filter(rule -> rule instanceof UsesEditorConfigProperties)
+				.flatMap(rule -> ((UsesEditorConfigProperties) rule).getEditorConfigProperties().stream());
+
+		// get complete list of supported properties in DefaultEditorConfigProperties.INSTANCE
+		List<EditorConfigProperty<?>> editorConfigProperties = new ArrayList<>(DefaultEditorConfigProperties.INSTANCE.getEditorConfigProperties());
+		editorConfigProperties.add(DefaultEditorConfigProperties.INSTANCE.getKtlintDisabledRulesProperty());
```
Comment (bcmedeiros) [inline lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java:102]: I think `ktlint_disabled_rules` has been added to `DefaultEditorConfigProperties` in 0.48.0, already, so this whole `new ArrayList` part would be useless now. I introduced it in https://github.com/diffplug/spotless/pull/1378 for a fix, but if it's in the `DefaultEditorConfigProperties` we can get rid of it.

Comment (nedtwigg) [reply lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java:102]: Thanks a ton @bcmedeiros for keeping an eye on this. @davidburstrom your call, I'm okay to merge it as-is, or if you want to make the change suggested above first I'm happy to wait.

Comment (brunojcm-vgw) [reply lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java:102]: Yeah, it definitely doesn't hurt.

Comment (davidburstrom) [reply lib/src/compatKtLint0Dot48Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot48Dot0Adapter.java:102]: I'll take a look at the suggested change.

**Comment by nedtwigg** [comment]
Needs a `spotlessApply`. First PR of 2023!

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 6.12.1` and `plugin-maven 2.29.0`.

**Comment by davidburstrom** [comment]
Thanks for doing the final cleanup!

**Pull request merged**